### ISSUE TITLE
`VcfWriter` does not write VCF index files anymore

### ISSUE THREAD

**Issue by mjhipp** [fulcrumgenomics/fgbio#812]
...unless overwriting an existing path (after #753)

When deciding whether to write an index, `VcfWriter` uses `isRegularFile()` on the output path. This will almost always return `false`, because the output path is usually a new, non-existing path (there is not already a regular file there).

Example (using `FilterSomaticVcf`), if I have this dir: 

```
.
├── existing-file.vcf.gz
├── input.bai
├── input.bam
├── input.vcf.gz
└── input.vcf.gz.tbi
```

And run the tool using a new output path:

```
❯ java -jar fgbio-1.5.1.jar FilterSomaticVcf -i input.vcf.gz -o new-output.vcf.gz -b input.bam
...
[2022/03/10 14:39:31 | FgBioMain | Info] FilterSomaticVcf completed. Elapsed time: 0.04 minutes.
```

I get the following output:

```
.
├── existing-file.vcf.gz
├── input.bai
├── input.bam
├── input.vcf.gz
├── input.vcf.gz.tbi
└── new-output.vcf.gz
```

If I instead write to the `existing-file.vcf.gz`, I get an index:

```
❯ java -jar fgbio-1.5.1.jar FilterSomaticVcf -i input.vcf.gz -o existing-file.vcf.gz -b input.bam
...
[2022/03/10 14:40:19 | FgBioMain | Info] FilterSomaticVcf completed. Elapsed time: 0.04 minutes.
```

I now get an index written:

```
.
├── existing-file.vcf.gz
├── existing-file.vcf.gz.tbi
├── input.bai
├── input.bam
├── input.vcf.gz
└── input.vcf.gz.tbi
```

Indexes are written using `1.4.0`:

```
❯ java -jar fgbio-1.4.0.jar FilterSomaticVcf -i input.vcf.gz -o new-output.vcf.gz -b input.bam
...
[2022/03/10 14:48:15 | FgBioMain | Info] FilterSomaticVcf completed. Elapsed time: 0.04 minutes.

❯ tree
.
├── input.bai
├── input.bam
├── input.vcf.gz
├── input.vcf.gz.tbi
├── new-output.vcf.gz
└── new-output.vcf.gz.tbi
```

**Comment by clintval** [comment]
That's my fault :/ I can take a stab at a fix soon. Would be nice to have this patched before 2.0

**Comment by tfenne** [comment]
I think we're still a week or more away from a 2.0 release, but I do want to cut a beta/milestone either tomorrow or Monday for folks to start testing
